azext_edge/edge/providers/rpsaas/adr/asset_endpoint_profiles.py

```diff
@@ -161,102 +213,150 @@ def update(
-            resource_group_name=resource_group_name
-        )
+# Helpers
+def _assert_above_min(param: str, value: int, minimum: int = 0) -> str:
+    if value < minimum:
+        return f"The parameter {param} needs to be at least {minimum}.\n"
+    return ""
 
-        return original_aep["properties"].get("transportAuthentication", {"ownCertificates": []})
 
-    def remove_transport_auth(
-        self,
-        asset_endpoint_profile_name: str,
-        thumbprint: str,
-        resource_group_name: str,
-    ):
-        original_aep = self.show(
-            asset_endpoint_profile_name,
-            resource_group_name=resource_group_name,
-            check_cluster_connectivity=True
+def _raise_if_connector_error(connector_type: str, error_msg: str):
+    if error_msg:
+        raise InvalidArgumentValueError(
+            f"The following {connector_type} connector arguments are invalid:\n {error_msg}"
         )
-        if original_aep["properties"].get("transportAuthentication") is None:
-            original_aep["properties"]["transportAuthentication"] = {
-                "ownCertificates": []
-            }
 
-        certs = original_aep["properties"]["transportAuthentication"]["ownCertificates"]
-        certs = [cert for cert in certs if cert["certThumbprint"] != thumbprint]
 
-        original_aep["properties"]["transportAuthentication"]["ownCertificates"] = certs
+def _build_opcua_config(
+    original_config: Optional[str] = None,
+    application_name: Optional[str] = None,
+    auto_accept_untrusted_server_certs: Optional[bool] = None,
+    default_publishing_interval: Optional[int] = None,
+    default_sampling_interval: Optional[int] = None,
+    default_queue_size: Optional[int] = None,
+    keep_alive: Optional[int] = None,
+    run_asset_discovery: Optional[str] = None,
+    session_timeout: Optional[int] = None,
+    session_keep_alive: Optional[int] = None,
+    session_reconnect_period: Optional[int] = None,
+    session_reconnect_exponential_back_off: Optional[int] = None,
+    security_policy: Optional[str] = None,
+    security_mode: Optional[str] = None,
+    sub_max_items: Optional[int] = None,
+    sub_life_time: Optional[int] = None,
+) -> str:
+    config = json.loads(original_config) if original_config else {}
+
+    error_msg = ""
+    if application_name:
+        config["applicationName"] = application_name
+    if keep_alive:
+        error_msg += _assert_above_min("--keep-alive", keep_alive)
+        config["keepAliveMilliseconds"] = keep_alive
+    if run_asset_discovery is not None:
+        config["runAssetDiscovery"] = run_asset_discovery
+
+    # defaults
+    if any([
+        default_publishing_interval, default_sampling_interval, default_queue_size
+    ]) and not config.get("defaults"):
+        config["defaults"] = {}
+    if default_publishing_interval:
+        error_msg += _assert_above_min("--default-publishing-int", default_publishing_interval, -1)
```

Comment:
What was the reason for not using the json schema directly? The json schema could be used to validate the incoming payload without needing to make custom validation logic.

Reply:
okie next pr

Reply:
wait let me put in a todo